### PR TITLE
feat(user): #325w2m6: add soft deleted for user

### PR DESCRIPTION
**Pull request by FremyXS**

### CODE REVIEW

File: UserManagementService.Api/UserManagementService.Api.csproj [hunk 1]
```diff
@@ -13,6 +13,7 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
     <PackageReference Include="Microsoft.VisualStudio.Azure.Containers.Tools.Targets" Version="1.17.0" />
+    <PackageReference Include="NodaTime" Version="3.1.4" />
```
Comment (Koval1997) [inline UserManagementService.Api/UserManagementService.Api.csproj]: Why we need to add this package to api dependencies?

File: UserManagementService.Core/Entities/User.cs [hunk 2]
```diff
@@ -13,6 +15,7 @@ public class User : IIdentityEntity
         public long RoleId { get; private set; }
 
         public Role Role { get; private set; }
+        public string? DeletedAtUtc { get; private set; }
```
Comment (Koval1997) [inline UserManagementService.Core/Entities/User.cs]: Type of this prop must be Date type from NodaTime package

File: UserManagementService.Core/Entities/User.cs [hunk 3]
```diff
@@ -41,5 +44,9 @@ public void AddRole(Role role)
         {
             Role = role;
         }
+        public void Delete(ZonedDateTime deletedAtUtc)
```
Comment (Koval1997) [inline UserManagementService.Core/Entities/User.cs]: We mustn't use ZonedDateTime here

File: UserManagementService.Application/Users/Commands/DeleteUserCommand.cs [hunk 4]
```diff
@@ -21,8 +22,10 @@ public DeleteUserCommandHandler(IUserRepository userRepository)
         public async Task Handle(DeleteUserCommand request)
         {
             var user = await _userRepository.FindByIdAsync(request.Id);
-
-            await _userRepository.RemoveAsync(user);
+            user.Delete(
+                SystemClock.Instance.GetCurrentInstant().InUtc()
```
Comment (Koval1997) [inline UserManagementService.Application/Users/Commands/DeleteUserCommand.cs]: We need to use https://nodatime.org/2.2.x/api/NodaTime.IClock.html for get current UTC time

File: UserManagementService.DataAccess/Respositories/InstantTime.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
+using NodaTime.Extensions;
+using NodaTime;
+using System;
+
+namespace UserManagementService.DataAccess.Respositories
+{
+    public class InstantTime : IClock
+    {
+        public Instant GetCurrentInstant()
```
Comment (Koval1997) [inline UserManagementService.DataAccess/Respositories/InstantTime.cs]: I think that we can rename from GetCurrentInstant to GetCurrentInstantInUtc

File: UserManagementService.Application/DependencyInjection.cs [hunk 6]
```diff
@@ -1,7 +1,9 @@
 using Microsoft.Extensions.DependencyInjection;
+using NodaTime;
 using UserManagementService.Application.Roles.Queries;
 using UserManagementService.Application.Users.Commands;
 using UserManagementService.Application.Users.Queries;
+using UserManagementService.DataAccess.Respositories;
```
Comment (Koval1997) [inline UserManagementService.Application/DependencyInjection.cs]: I think that we cannot use DataAccess layer in application layer. And why we store InstantTime in DataAccess layer?

File: UserManagementService.DataAccess/Respositories/InstantTime.cs [hunk 7]
```diff
@@ -0,0 +1,14 @@
+using NodaTime.Extensions;
+using NodaTime;
+using System;
+
+namespace UserManagementService.DataAccess.Respositories
+{
+    public class InstantTime : IClock
```
Comment (Koval1997) [inline UserManagementService.DataAccess/Respositories/InstantTime.cs]: Change name from InstantTime to simple Clock

File: UserManagementService.DataAccess/Respositories/InstantTime.cs [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+    {
+        public Instant GetCurrentInstant()
+        {
+            return DateTime.UtcNow.ToInstant();
```
Comment (Koval1997) [inline UserManagementService.DataAccess/Respositories/InstantTime.cs]: Need to use Instant.FromDateTimeUtc(DateTime.UtcNow)

File: UserManagementService.DataAccess/Respositories/InstantTime.cs (hunk 5 above)
Comment (FremyXS) [reply UserManagementService.DataAccess/Respositories/InstantTime.cs]: The method is inherited from the interface

**Pull request merged**